stellargraph/layer/gcn_lstm.py

```diff
@@ -0,0 +1,347 @@
+        }
+
+        base_config = super().get_config()
+        return {**base_config, **config}
+
+    def calculate_laplacian(self, adj):
+        D = np.diag(np.ravel(adj.sum(axis=0)) ** (-0.5))
+        adj = np.dot(D, np.dot(adj, D))
+        return adj
+
+    def compute_output_shape(self, input_shapes):
+        """
+        Computes the output shape of the layer.
+        Assumes the following inputs:
+
+        Args:
+            input_shapes (tuple of ints)
+                Shape tuples can include None for free dimensions, instead of an integer.
+
+        Returns:
+            An input shape tuple.
+        """
+        feature_shape = input_shapes
+
+        return feature_shape[0], feature_shape[1], self.units
+
+    def set_A(self, A):
+        """
+        Sets the adjacnency matrix provided as input.
+        
+        """
+        K.set_value(self.A, A)
+
+    def build(self, input_shapes):
+        """
+        Builds the layer
+
+        Args:
+            input_shapes (list of int): shapes of the layer's inputs (node features and adjacency matrix)
+
+        """
+        n_nodes = input_shapes[-1]
+        t_steps = input_shapes[-2]
+        self.units = t_steps
+        
+        self.A = self.add_weight(
+            name="A",
+            shape=(n_nodes, n_nodes),
+            trainable=False,
+            initializer=initializers.constant(self.adj),
+        )
+
+        # K.set_value(self.A, self.adj)
+
+        self.kernel = self.add_weight(
+            shape=(t_steps, self.units),
+            initializer=self.kernel_initializer,
+            name="kernel",
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+        )
+
+        if self.use_bias:
+            self.bias = self.add_weight(
+                shape=(n_nodes,),
+                initializer=self.bias_initializer,
+                name="bias",
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+            )
+        else:
+            self.bias = None
+        self.built = True
+
+    def call(self, features):
+        """
+        Applies the layer.
+
+        Args:
+            inputs (ndarray): node features (size B x T x N),
+                where B is the batch size,
+                      T is the sequence length, and
+                      N is the number of nodes in the graph.
+
+        Returns:
+            Keras Tensor that represents the output of the layer.
+        """
+
+        # Calculate the layer operation of GCN
+
+        h_graph = K.dot(features, self.A)
+        output = tf.transpose(
+            K.dot(tf.transpose(h_graph, [0, 2, 1]), self.kernel), [0, 2, 1]
+        )
+
+        # Add optional bias & apply activation
+        if self.bias is not None:
+            output += self.bias
+
+        output = self.activation(output)
+
+        return output
+
+
+class Graph_Convolution_LSTM(Model):
```

Comment:
Our other "models" aren't actually formal models, meaning they don't inherit from `tf.keras.Model`. I think it would be good to stick to that.

Also, Python style is to not have underscores in class names, e.g.

```suggestion
class GraphConvolutionLSTM(Model):
```

but potentially it's best to break the "style" slightly and call it:

```suggestion
class GCN_LSTM(Model):
```